runtime/src/accounts_db.rs

```diff
@@ -808,16 +810,32 @@ impl AccountsDB {
         Self::hash_account_data(
             slot,
             account.account_meta.lamports,
+            account.account_meta.executable,
+            account.account_meta.rent_epoch,
             account.data,
             &account.meta.pubkey,
         )
     }
 
     pub fn hash_account(slot: Slot, account: &Account, pubkey: &Pubkey) -> Hash {
-        Self::hash_account_data(slot, account.lamports, &account.data, pubkey)
+        Self::hash_account_data(
+            slot,
+            account.lamports,
+            account.executable,
+            account.rent_epoch,
+            &account.data,
+            pubkey,
+        )
     }
 
-    pub fn hash_account_data(slot: Slot, lamports: u64, data: &[u8], pubkey: &Pubkey) -> Hash {
+    pub fn hash_account_data(
+        slot: Slot,
+        lamports: u64,
+        executable: bool,
+        rent_epoch: Epoch,
+        data: &[u8],
+        pubkey: &Pubkey,
+    ) -> Hash {
         if lamports == 0 {
```

Comment:
Well, this hash algorithm seems littered with many modifications...  #6315 #5537 #3801